click/testing.py

```diff
@@ -292,14 +294,19 @@ def invoke(self, cli, args=None, input=None, env=None,
 
                 exit_code = e.code
                 if not isinstance(exit_code, int):
+                    if exit_code is None:
+                        exit_code = 0
+                    else:
+                        exit_code = 1
```

Comment:
This breaks the usecase where somebody calls `sys.exit("Error message")`. This is why the `exit_code` was overridden *after* we print the message